chef-powershell/lib/chef-powershell/powershell.rb

```diff
@@ -83,24 +99,31 @@ def self.set_ps_timeout(value)
 
       def self.do_work
         ffi_lib @@powershell_dll
-        attach_function :execute_powershell, :ExecuteScript, %i{string int}, :pointer
-        execute_powershell(@@ps_command, @@ps_timeout)
+        attach_function :execute_powershell, :ExecuteScript, %i{string int pointer}, :pointer
+
+        execute_powershell(@@ps_command, @@ps_timeout, AllocateCallback)
       end
     end
 
     private
 
     def exec(script, timeout: -1)
       timeout = -1 if timeout == 0 || timeout.nil?
+
```

Comment:
Oh I think I see. The above takes care of that default issue?

Reply:
Multiple layers do, all the way up to the entry point into the gem, which did not change (`ChefPowerShell::ChefPowerShellModule::PowerShellExec.powershell_exec` module which gets `include`d into resources and other classes on the Chef Client side.